Plugin Vault/src/main/java/net/simon987/vaultplugin/VaultDoor.java

```diff
@@ -19,77 +20,148 @@
      */
     private char[] password;
 
+    private RandomString random_string_generator;
+
     /**
      * Whether or not the vault door is opened
      */
-    private boolean opened;
+    private boolean open = false;
 
-    private int openedTimer;
 
     /**
      * Number of ticks to remain the door open
      */
-    private static final int OPEN_TIME = 4; //todo load from config
+    private int OPEN_TIME = GameServer.INSTANCE.getConfig().getInt("vault_door_open_time");
+    private int openedTimer = 0;
 
+    private int password_length;
```

Comment:
Password length should always be equal to ComPort.MESSAGE_LENGTH (which is 8). This would simplify a lot the programming: if the password is longer than 8 characters, there must be multiple ComPort `COMPORT_FRONT_PORT_OUT` operations. How do you keep track of them? When do you know which one is the last `COMPORT_FRONT_PORT_OUT` operation and how would it work when multiple players are sending messages to the door? There is plenty of possible combinations for a fixed length of 8 characters

Reply:
Agreed: 8 alphanumeric characters give little short of 48 bits of key, which is plenty hard to brute-force.